packages/app-desktop/gui/ProfileManagementScreen.tsx

```diff
@@ -0,0 +1,277 @@
+
+interface Props {
+	themeId: number;
+	style: CSSProperties;
+	// eslint-disable-next-line @typescript-eslint/ban-types -- Changing types for these variables would be too big of a refactoring
+	dispatch: Function;
+	profileConfig: ProfileConfig;
+}
+
+interface State {
+	profiles: Profile[];
+	filter: string;
+}
+
+interface ProfileTable {
+	profiles: Profile[];
+	currentProfileId: string;
+	onProfileRename: (profile: Profile)=> void;
+	onProfileDelete: (profile: Profile)=> void;
+	filter: string;
+	themeId: number;
+	style: CSSProperties;
+}
+
+const ProfileTableComp = (props: ProfileTable) => {
+	const theme = themeStyle(props.themeId);
+
+	const nameCellStyle = {
+		...theme.textStyle,
+		textOverflow: 'ellipsis',
+		overflowX: 'hidden',
+		maxWidth: 1,
+		width: '100%',
+		whiteSpace: 'nowrap',
+	};
+
+	const cellStyle = {
+		...theme.textStyle,
+		whiteSpace: 'nowrap',
+		color: theme.colorFaded,
+		width: 1,
+	};
+
+	const headerStyle = {
+		...theme.textStyle,
+		whiteSpace: 'nowrap',
+		width: 1,
+		fontWeight: 'bold',
+	};
+
+	const filteredProfiles = props.profiles.filter(
+		(profile: Profile) => !props.filter || profile.name?.toLowerCase().includes(props.filter.toLowerCase()) || profile.id.includes(props.filter),
+	);
+
+	return (
+		<table style={{ width: '100%' }}>
+			<thead>
+				<tr>
+					<th style={headerStyle}>{_('Profile name')}</th>
+					<th style={headerStyle}>{_('ID')}</th>
+					<th style={headerStyle}>{_('Status')}</th>
+					<th style={headerStyle}>{_('Actions')}</th>
+				</tr>
+			</thead>
+			<tbody>
+				{filteredProfiles.map((profile: Profile, index: number) => {
+					const isCurrentProfile = profile.id === props.currentProfileId;
+					return (
+						<tr key={index}>
+							<td id={`name-${profile.id}`} style={nameCellStyle} className="nameCell">
+								<span style={{ fontWeight: isCurrentProfile ? 'bold' : 'normal' }}>
+									{profile.name || `(${_('Untitled')})`}
+								</span>
+							</td>
+							<td style={cellStyle} className="dataCell">{profile.id}</td>
+							<td style={cellStyle} className="dataCell">
+								{isCurrentProfile ? _('Active') : ''}
+							</td>
+							<td style={cellStyle} className="dataCell">
+								<button
+									id={`rename-${profile.id}`}
+									aria-labelledby={`rename-${profile.id} name-${profile.id}`}
+									style={{ ...theme.buttonStyle, marginRight: 10 }}
+									onClick={() => props.onProfileRename(profile)}
+								>
+									{_('Rename')}
+								</button>
+								{!isCurrentProfile && (
+									<button
+										id={`delete-${profile.id}`}
+										aria-labelledby={`delete-${profile.id} name-${profile.id}`}
+										style={theme.buttonStyle}
+										onClick={() => props.onProfileDelete(profile)}
+									>
+										{_('Delete')}
+									</button>
+								)}
+							</td>
+						</tr>
+					);
+				})}
+			</tbody>
+		</table>
+	);
+};
+
+class ProfileManagementScreenComponent extends React.Component<Props, State> {
```

Comment:
Style: Please [use function components for new code.](https://joplinapp.org/help/dev/coding_style#use-function-components-for-new-code)